### ISSUE TITLE
4.0-dev branch

### ISSUE THREAD

**Issue by BastiaanOlij** [GodotVR/godot_openxr#91]
I've created a 4.0-dev branch to start the porting work to Godot 4.0 
This will likely be a branch that won't work for the foreseeable future as we're waiting on a number of upstream changes in Godot to be accepted but I'm trying to get a head start in this work.

I'll also do my best to try and keep this branch rebase on master.

**Comment by BastiaanOlij** [comment]
Hi @asampal ,

We're in the process of porting this logic into the core, it's working for PCVR now with just a few features left to port but not merged yet:
https://github.com/godotengine/godot/pull/56394

You can compile that PR and play around with it.

**Comment by m4gr3d** [comment]
Now that initial support has been merged into `core`, we should delete the `4.0-dev` branch.

**Comment by BastiaanOlij** [comment]
Agreed!
